ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/TransactionsReader.scala

```diff
@@ -562,11 +562,16 @@ private[appendonlydao] object TransactionsReader {
       s"You can only split a range in a strictly positive number of chunks ($numberOfChunks)",
     )
 
-    val minChunkSize = math.max(1, maxChunkSize / 10)
-
     val rangeSize = endInclusive - startExclusive
+    require(
+      rangeSize >= 0,
+      s"Range size should be positive but got bounds ($startExclusive, $endInclusive]",
+    )
+
+    val minChunkSize = math.max(1, maxChunkSize / 10)
 
-    if (rangeSize == 0L) Vector.empty
+    if (rangeSize <= 0L)
```

Comment:
Changing this condition is not required as we have the `require()` above but it's fine for me to keep it this way :)

Reply:
Right, changed.

Reply:
Perfect, thanks!